Zend/zend_compile.c

```diff
@@ -7904,6 +7905,12 @@ static void zend_compile_func_decl(znode *result, zend_ast *ast, bool toplevel)
 		zend_compile_closure_uses(uses_ast);
 	}
 
+	deprecated = zend_get_attribute_str(op_array->attributes, "deprecated", sizeof("deprecated")-1);
+
+	if (deprecated && stmt_ast != NULL) {
```

Comment:
What is the `stmt_ast != NULL` check about? Is it just to avoid adding `#[Deprecated]` to abstract functions that cannot be called? Shouldn't they still be reflectable as deprecated?

Reply:
I'll check with @beberlei. This part is code written by him.

Reply:
:point_right: https://externals.io/message/123397#123568

Reply:
No comments on the list, that likely means that everyone is happy. I've now removed the `stmt_ast` check.

Reply:
Sounds good. I was planning to respond to the list, but you know how it is. ^^ I think this approach (to allow and respect `#[Deprecated]` for reflection only) makes sense.